src/Filesystem/Dir.php

```diff
@@ -139,23 +139,31 @@ public static function files(
 
 	/**
 	 * Read the directory and all subdirectories
+	 *
+	 * @todo Remove support for `$ignore = null` in a major release
+	 * @param array|null $ignore Passing null is deprecated.
 	 */
 	public static function index(
 		string $dir,
 		bool $recursive = false,
-		array|null $ignore = [],
+		array|bool|null $ignore = [],
 		string $path = null
 	): array {
 		$result = [];
 		$dir    = realpath($dir);
-		$items  = static::read($dir);
+		$items  = static::read($dir, $ignore === false ? [] : null);
+		$ignore = (array)$ignore;
```

Comment:
If `$ignore` is `false` this will create `[false]`

Reply:
I think we can remove this line completely as we check for `is_array()` below anyway.